### DISCUSSION TITLE
Custom hit range

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#838] [Help] [answered]
Hey, is it possible to create custom hit range? Like 2 blocks, 3 blocks and even 4 blocks?

**Comment by MrGazdag** [comment, marked as answer]
You can check the distance to the entity with raycasting: find the point at which the cursor vector intersects with the bounding box of the entity, and then checking the length of the ray. 
You cannot modify the client’s range though, so attack/interact events will not fire for ranges over the default value. However, if you manually raycast and check entities on left clicks, you can achieve larger range.
